src/modules/CurrencyModule.cpp

```diff
@@ -1039,24 +1042,47 @@ CurrencyModule::CurrencyModule(UmikoBot* client) : Module("currency", true), m_c
 				client.createMessage(message.channelId(), "**I can't let you make your victim go into serious debt.**");
 				return;
 			}
+
+			// https://www.desmos.com/calculator/z6b0k7wb1t
+			// This success chance is in the range of 0 to 1
+			double successChance = (config.stealSuccessChance / 100.0) * qExp(-0.0001 * qPow(amountToSteal, 1.5));
+			
+			QString thiefName = UmikoBot::Instance().GetName(channel.guildId(), authorId);
+			QString victimName = UmikoBot::Instance().GetName(channel.guildId(), victimId);
+		
+			std::random_device device;
+			std::mt19937 prng { device() };
+			std::discrete_distribution<> distribution { { 1 - successChance, successChance } };
+			int roll = distribution(prng);
 			
-			if (qrand() % 100 < config.stealSuccessChance)
+			if (roll)
```

Comment:
Can you explain how you don't have to compare against some target chance any more, and just check if the integer is non-zero?